metagpt/config.py

```diff
@@ -114,5 +120,19 @@ def get(self, key, *args, **kwargs):
             raise ValueError(f"Key '{key}' not found in environment variables or in the YAML file")
         return value
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        CONFIGS_CTX.get()[name] = value
```

Comment:
why not CONFIGS_CTX.set ?

Reply:
The value of the CONTEXTS_CTX context variable is a dictionary used to store all configuration variables, and the \_\_setattr\_\_ method is used to set new configuration variables, thereby adding them to the dictionary. CONFIGS_CTX.set is used to set a new dictionary value in the context, and this method is often called during coroutine initialization.